tests/unittests/library/test_nsx_common.py

```diff
@@ -628,6 +637,151 @@ def test_nsx_read(self):
             'c'
         )
 
+    @pytest.mark.internal
+    @pytest.mark.unit
+    def test_nsx_search(self):
```

Comment:
This should be split into 5 tests with more helpful names (so that if one fails others might still succeed so that whoever was maintaining it and caused the problem would know specifically what the issue was).

e.g. test_nsx_search_bad_http_result, etc